packages/cli/internal/pkg/aws/cdk/deploy_app.go

```diff
@@ -8,7 +8,7 @@ import (
 
 var mkDirTemp = os.MkdirTemp
 
-func (client Client) DeployApp(appDir string, context []string) (ProgressStream, error) {
+func (client Client) DeployApp(appDir string, context []string, uniqueKey string) (ProgressStream, error) {
```

Comment:
This `uniqueKey` argument feels out of place, can we consider other options? 
- use context to propagate this unique key (we can also use context for timeouts)
- auto-generate this unique key